### DISCUSSION TITLE
How to remove codelldb from the clangd lang extra?

### DISCUSSION THREAD

**Discussion by sumeetgadagkar** [LazyVim/LazyVim#5085] [Q&A] [answered]
I enabled the clangd lang. extra but I do not want to use codelldb. How can I remove it from the extra?

I have tried : 

```lua
return {
    {
        "williamboman/mason.nvim",
        opts = function(_, opts)
            if opts.ensure_installed then
                opts.ensure_installed = vim.tbl_filter(function(item)
                    return item ~= "codelldb"
                end, opts.ensure_installed)
            end
        end,
    },
},

return {
    {
        "jay-babu/mason-nvim-dap.nvim",
        opts = function(_, opts)
            if opts.ensure_installed then
                opts.ensure_installed = vim.tbl_filter(function(item)
                    return item ~= "codelldb"
                end, opts.ensure_installed)
            end
        end,
    },
}
```

But every time I add a breakpoint, I get the message `mason-nvim-dap : automatically installing codelldb`. Any idea how to do this? I would really like to avoid disabling the extra and copying the configs in extra to my own config. Thanks in advance!

**Comment by dpetka2001** [comment, marked as answer]
I believe that's because in `jay-babu/mason-nvim-dap.nvim` the option `automatic_installation = true` is set. Try setting that to `false` otherwise if it sees a dap configuration for a debugger and the debugger is not installed, it will try to install it. Do that along with the modifications you did in `ensure_installed` to not get installed as a dependency from Mason.

**Reply by sumeetgadagkar** [reply]
Thanks that helped!
